internal/ocs/validations.go

```diff
@@ -0,0 +1,81 @@
+package ocs
+
+import (
+	"encoding/json"
+	"math"
+
+	"github.com/openshift/assisted-service/internal/host"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+//Info is used to return the parameters required for OCS Validations
+type Info struct {
+	CountOfOcsDisks      int64
+	CanDeployOcs         bool
+	OcsMinimalDeployment bool
+	StorageCapacity      int64
+}
+
+// ValidateOCSRequirements is used to validate min requirements of OCS
+func ValidateOCSRequirements(log logrus.FieldLogger, cluster *models.Cluster, h host.API) *Info {
+	hosts := cluster.Hosts
+	if len(hosts) < 3 {
+		log.Errorf("Insufficient hosts to deploy OCS")
+		return &Info{
+			CanDeployOcs: false,
+		}
+	}
+	var CPUCount int64
+	var TotalRAM int64
+	var DiskCount int64
+	var MinDiskSize int64 // min disk size after boot disk size to claim for the PV created by localVolume
+	MinDiskSize = math.MaxInt64
+	for _, host := range hosts {
+		var inventory models.Inventory
+		if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+			log.Errorf("Failed to get inventory from host with id %s", host.ID)
+			return &Info{
+				CanDeployOcs: false,
+			}
+		}
+		//	disks, err := hwValidator.GetHostValidDisks(host)
+		disks, err := h.GetHostValidDisks(host)
+		if err != nil {
+			log.Error("failed to get valid disks due to ", err.Error())
+			return &Info{
+				CanDeployOcs: false,
+			}
+		}
+		if len(disks) > 1 {
+
+			MinDiskSize = (int64)(math.Min(float64(MinDiskSize), float64(disks[1].SizeBytes)))
+		}
+		CPUCount += inventory.CPU.Count
+		TotalRAM += inventory.Memory.UsableBytes
+		DiskCount += (int64)(len(inventory.Disks) - 1)
+	}
+	if DiskCount < 3 {
+		log.Errorf("Insufficient disk count")
+		return &Info{
+			CanDeployOcs: false,
+		}
+	} else if CPUCount < 24 || TotalRAM < 66000000000 { // insuficient resources
+		log.Errorf("Insufficient CPU count or TotalRAM")
+		return &Info{
+			CanDeployOcs: false,
+		}
+	} else if CPUCount < 30 || TotalRAM < 72000000000 { // requires OCS deployment with minimal resources
+		return &Info{
+			CanDeployOcs:         true,
+			OcsMinimalDeployment: true,
```

Comment:
I thought "minimal deployment" was an option that the user could set via API, and then this code would set CanDeployOcs to true or false depending on the CPUCount and TotalRAM, together with whether the user set minimal deployment.

Reply:
Are we allowing the user to choose? I thought it was decided on the basis of CPUCount and RAM .